pkg/apis/apps.kubermatic/v1/application_definition.go

```diff
@@ -35,27 +35,39 @@ const (
 type HelmCredentials struct {
 	// Username holds the ref and key in the secret for the username credential.
 	// The Secret must exist in the namespace where KKP is installed (default is "kubermatic").
-	// The Secret must be annotated with `apps.kubermatic.k8c.io/secret-type:` set to helm or git
+	// The Secret must be annotated with `apps.kubermatic.k8c.io/secret-type:` set to "helm" or "git"
 	Username *corev1.SecretKeySelector `json:"username,omitempty"`
 
-	// Password holds the ref and key in the secret for the Password credential.
+	// Password holds the ref and key in the secret for the password credential.
 	// The Secret must exist in the namespace where KKP is installed (default is "kubermatic").
-	// The Secret must be annotated with `apps.kubermatic.k8c.io/secret-type:` set to helm or git
+	// The Secret must be annotated with `apps.kubermatic.k8c.io/secret-type:` set to "helm" or "git"
 	Password *corev1.SecretKeySelector `json:"password,omitempty"`
 
-	// RegistryConfigFile holds the ref and key in the secret for the registry credential file. The value is dockercfg
-	// file that follows the same format rules as ~/.docker/config.json
-	// The The Secret must exist in the namespace where KKP is installed (default is "kubermatic").
-	// The Secret must be annotated with `apps.kubermatic.k8c.io/secret-type:` set to helm or git
+	// RegistryConfigFile holds the ref and key in the secret for the registry credential file.
+	// The value is dockercfg file that follows the same format rules as ~/.docker/config.json.
+	// The Secret must exist in the namespace where KKP is installed (default is "kubermatic").
+	// The Secret must be annotated with `apps.kubermatic.k8c.io/secret-type:` set to "helm" or "git"
 	RegistryConfigFile *corev1.SecretKeySelector `json:"registryConfigFile,omitempty"`
 }
 
 type HelmSource struct {
-	// URl of the helm repository.
-	// It can be an HTTP(s) repository (e.g. https://localhost/myrepo) or on OCI repository (e.g. oci://localhost:5000/myrepo).
 	// +kubebuilder:validation:Pattern="^(http|https|oci)://.+"
+
+	// URL of the Helm repository the following schemes are supported:
+	//
+	// * http://localhost/myrepo (HTTP)
+	// * https://localhost/myrepo (HTTPS)
+	// * oci://localhost:5000/myrepo (OCI, HTTPS or HTTP depending on the useHTTP flag)
 	URL string `json:"url"`
 
+	// Insecure disables certificate validation when using an HTTPS registry. This setting has no
+	// effect when using a plaintext connection.
+	Insecure *bool `json:"insecure,omitempty"`
+
+	// UseHTTP will enable HTTP-only (i.e. unencrypted) traffic for oci:// URLs. By default HTTPS
+	// is used when communicating with an oci:// URL.
+	UseHTTP *bool `json:"useHTTP,omitempty"`
```

Comment:
Nit: I would prefer to name this `PlainHTTP` to be consistent with how helm calls it https://github.com/helm/helm/blob/d58d7b376265338e059ff11c71267b5a6cf504c3/cmd/helm/flags.go#L64

It's not the best name I know, but I like the consistency to have it easier for our end-users to follow